templates/contrib/jdbc/legacy/build.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+set -ueo pipefail
+
+if [ -n "${DEBUG:-}" ] ; then
+    set -x
+fi
+
+function print_help() {
+    echo "----------------------------------------------------------"
+    echo "Builds and pushes JDBC Driver images to a docker registry. This has to be executed from a folder containing a Dockerfile"
+    echo ""
+    echo "Usage: "
+    echo "   ../build.sh [--registry=myregistry.example.com:5000] [--artifact-repo=https://myrepo.example.com/maven/public]"
+    echo "Options:"
+    echo "   Available driver images to build: db2, mariadb, mssql, mysql, oracle, postgresql, sybase"
+    echo "   --registry         Specifies the docker registry to use for tagging and pushing. Defaults to docker-registry.default.svc:5000"
+    echo "   --artifact-repo    Specifies the Maven repository where the jdbc drivers are available. Oracle does not have a default value"
+    echo "   --image-tag        Specifies the tag to use when building the image. Defaults to 1.0"
+}
+
+while (($#))
+do
+    case $1 in
+        --registry=*)
+            registry=${1#*=}
+        ;;
+        --artifact-repo=*)
+            artifact_repo=${1#*=}
+        ;;
+        --image-tag=*)
+            image_tag=${1#*=}
+        ;;
+        -h)
+            print_help
+            exit 0
+        ;;
+        --help)
+            print_help
+            exit 0
+        ;;
+    esac
+shift
+done
+
+if [[ ! -f Dockerfile ]]
+then
+    echo "Error: No Dockerfile found"
+    print_help
+    exit 1
+fi
+
+current_dir=${PWD##*/}
+driver=$(echo $current_dir | cut -d '-' -f 1)
+image_tag=${image_tag:-1.0}
+
+registry=${registry:-docker-registry.default.svc:5000}
+
+function docker_login() {
+    if [[ $(oc whoami -t > /dev/null; echo $?) == 1 ]]; then
+        echo "You must be logged in"
+        exit 1
+    fi
+    docker login -u `oc whoami` -p `oc whoami -t` $registry
```

Comment:
The problem with this docker login is that ties up with the Openshift registry and also does not work fine with registries with invalid certificates.

I tried to run this script but our QE Openshift instances do not have a registry with a valid certificate (docker cannot handle this, podman does using --tls-verify=false).

I think we could remove the docker_login from this script and do the login before running this script as done for logging using the OC client.
What do you think?

Reply:
it was mostly copy/paste from the previous version, so yes, I think it is reasonable.